roachtest: activerecord failed

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7814869?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/7814869?buildTab=artifacts#/activerecord) on master @ [9c21578450e395c83a1dc0df7090296fef06e006](https://github.com/cockroachdb/cockroach/commits/9c21578450e395c83a1dc0df7090296fef06e006):


```
--- FAIL: HasManyThroughAssociationsTest#test_has_many_association_through_a_has_many_association_with_nonstandard_primary_keys (unexpected)
--- FAIL: UpdateAllTest#test_update_counters_cares_about_optimistic_locking (unexpected)
--- FAIL: DDLTest#test_add_geometry_column (unexpected)
--- FAIL: NestedThroughAssociationsTest#test_has_many_through_belongs_to_with_has_many_through_source_reflection_preload_via_joins (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_with_create_exclamation_and_no_options (unexpected)
--- FAIL: HasOneThroughAssociationsTest#test_creating_association_builds_through_record (unexpected)
--- FAIL: QueryCacheTest#test_cache_is_available_when_using_a_not_connected_connection (unexpected)
--- FAIL: FixturesTest#test_instantiation (unexpected)
--- FAIL: LegacyPolymorphicReferenceIndexTest::V5_0#test_polymorphic_add_reference_uses_all_column_names_in_index (unexpected)
--- FAIL: NestedThroughAssociationsTest#test_has_one_through_has_one_with_has_one_through_source_reflection (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_preload_sti_middle_relation (unexpected)
--- FAIL: FoxyFixturesTest#test_identifies_consistently (unexpected)
--- FAIL: TestDestroyAsPartOfAutosaveAssociation#test_should_destroy_has_many_as_part_of_the_save_transaction_if_they_were_marked_for_destruction (unexpected)
--- FAIL: ActiveRecord::CockroachDB::Migration::ForeignKeyTest#test_add_invalid_foreign_key (unexpected)
--- FAIL: NestedThroughAssociationsTest#test_has_many_through_has_one_through_with_has_many_source_reflection_preload_via_joins (unexpected)
--- FAIL: TestAutosaveAssociationOnAHasManyAssociation#test_should_not_update_children_when_parent_creation_with_no_reason (unexpected)
--- FAIL: FoxyFixturesTest#test_generates_unique_ids (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_include? (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_collection_singular_ids_through_setter_raises_exception_when_invalid_ids_set (unexpected)
--- FAIL: TimestampTest#test_no_touching_threadsafe (unexpected)
--- FAIL: ViewWithoutPrimaryKeyTest#test_column_definitions (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_marshal_dump (unexpected)
--- FAIL: TestDestroyAsPartOfAutosaveAssociation#test_should_skip_validation_on_has_many_if_marked_for_destruction (unexpected)
--- FAIL: ActiveRecord::ConnectionAdapters::ConnectionHandlersShardingDbTest#test_sharding_separation (unexpected)
--- FAIL: TestDestroyAsPartOfAutosaveAssociation#test_a_child_marked_for_destruction_should_not_be_destroyed_twice_while_saving_habtm (unexpected)
--- FAIL: ActiveRecord::CockroachDB::Migration::ForeignKeyTest#test_add_foreign_key_inferes_column (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_belongs_to_with_dependent_destroy (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnTest#test_change_column_of_parent_table (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_create_on_new_record (unexpected)
--- FAIL: ActiveRecord::Migration::ForeignKeyChangeColumnTest#test_rename_column_of_child_table (unexpected)
--- FAIL: DDLTest#test_column_defaults (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_create_record (unexpected)
--- FAIL: NestedThroughAssociationsTest#test_has_many_through_has_and_belongs_to_many_with_has_many_source_reflection (unexpected)
--- FAIL: ViewWithoutPrimaryKeyTest#test_attributes (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_with_create_bang_and_invalid_options (unexpected)
--- FAIL: TestDestroyAsPartOfAutosaveAssociation#test_should_destroy_habtm_as_part_of_the_save_transaction_if_they_were_marked_for_destruction (unexpected)
--- FAIL: HasOneThroughAssociationsTest#test_association_create_constructor_creates_through_record (unexpected)
--- FAIL: QueryCacheTest#test_query_cache_with_multiple_handlers_and_forked_processes_legacy_handling (unexpected)
--- FAIL: TestDestroyAsPartOfAutosaveAssociation#test_should_skip_validation_on_habtm_if_destroyed (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_associate_existing_record_twice_should_add_records_twice (unexpected)
--- FAIL: FoxyFixturesTest#test_automatically_sets_primary_key (unexpected)
--- FAIL: TimestampTest#test_touching_a_record_without_timestamps_is_unexceptional (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_update_counter_caches_on_destroy_with_indestructible_through_record (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_through_association_readonly_should_be_false (unexpected)
--- FAIL: QueryCacheTest#test_cache_does_not_raise_exceptions (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_joining_has_many_through_belongs_to (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_delete_all_for_with_dependent_option_destroy (unexpected)
--- FAIL: FixturesTest#test_serialized_fixtures (unexpected)
For a full summary look at the activerecord artifacts
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_fs=ext4</code>
, <code>ROACHTEST_localSSD=true</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89367 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.2.0]
- #87580 roachtest: activerecord failed [C-test-failure O-roachtest O-robot branch-release-22.2]
- #82805 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.1]
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-22073

--- a/pkg/cmd/roachtest/tests/activerecord_blocklist.go
+++ b/pkg/cmd/roachtest/tests/activerecord_blocklist.go
@@ -33,9 +33,7 @@ var activeRecordBlocklist = blocklist{
 	"PostgresqlRangeTest#test_timezone_array_awareness_tsrange":                                                                         "27791",
 	"PostgresqlRangeTest#test_timezone_array_awareness_tzrange":                                                                         "27791",
 	// TODO(yang): investigate cause further
-	"PostgreSQLReferentialIntegrityTest#test_all_foreign_keys_valid_having_foreign_keys_in_multiple_schemas":                             "unknown",
-	"PostgresqlTimestampWithAwareTypesTest#test_timestamp_with_zone_values_with_rails_time_zone_support_and_time_zone_set":               "expected ActiveSupport::TimeWithZone, not Time",
-	"PostgresqlTimestampWithTimeZoneTest#test_timestamp_with_zone_values_with_rails_time_zone_support_and_timestamptz_and_time_zone_set": "expected ActiveSupport::TimeWithZone, not Time",
+	"PostgreSQLReferentialIntegrityTest#test_all_foreign_keys_valid_having_foreign_keys_in_multiple_schemas": "unknown",
 }
 
 var activeRecordIgnoreList = blocklist{
@@ -73,6 +71,8 @@ var activeRecordIgnoreList = blocklist{
 	"LengthValidationTest#test_validates_size_of_association_using_within":                                                                                     "flaky - sometimes complains that a relation does not exist",
 	"PostgresqlInfinityTest#test_where_clause_with_infinite_range_on_a_datetime_column":                                                                        "flaky - sometimes complains that a relation does not exist",
 	"PostgresqlIntervalTest#test_interval_type":                                                                                                                "flaky",
+	"PostgresqlTimestampWithAwareTypesTest#test_timestamp_with_zone_values_with_rails_time_zone_support_and_time_zone_set":                                     "flaky - sometimes complains given Time instead of ActiveSupport::TimeWithZone",
+	"PostgresqlTimestampWithTimeZoneTest#test_timestamp_with_zone_values_with_rails_time_zone_support_and_timestamptz_and_time_zone_set":                       "flaky - sometimes complains given Time instead of ActiveSupport::TimeWithZone",
 	"RelationTest#test_finding_last_with_arel_order":                                                                                                           "flaky - sometimes complains that a relation does not exist",
 	"RelationTest#test_joins_with_select_custom_attribute":                                                                                                     "flaky - sometimes complains that a relation does not exist",
 	"RelationTest#test_order_triggers_eager_loading_when_ordering_using_hash_syntax":                                                                           "flaky - sometimes complains that a relation does not exist",

